[SYCL][Opaque pointer] Enable INTEL_SYCL_OPAQUEPOINTER_READY

jsji opened the pull request:
This is to enable -DINTEL_SYCL_OPAQUEPOINTER_READY=1 by default, 
so that we can start to remove the old code already removed by community (those guarded within #ifndef INTEL_SYCL_OPAQUEPOINTER_READY ).

- Set opauqe pointer ready
- Add flag to tablegen
- Remove opaque pointer from merge-extension-ivars.m
- Port barrier_intrinsic.ll to opaque pointer
- Remove intel-alloca-bitcast.ll as it is testing typed pointer path
- Revert changes to InferAddressSpaces/AMDGPU/icmp.ll
- Use private repo for testing purpose

jsji commented:
Ping @intel/dpcpp-clang-driver-reviewers @intel/dpcpp-esimd-reviewers @intel/dpcpp-spirv-reviewers @intel/dpcpp-tools-reviewers . 

Most in these components are just test update, please have a look. Thanks.
The plan is to merge this before EOD tomorrow. Thanks.

sarnex reviewed (approved):
esimd lgtm

asudarsa commented on llvm/lib/SYCLLowerIR/CMakeLists.txt:
```diff
@@ -12,11 +12,8 @@ endif()
-    # Date:   Tue Jun 6 23:56:43 2023 +0000
-    # Mark llvm15 tests failures
-    set(LLVMGenXIntrinsics_GIT_TAG 910db4801d4a029834606e3e42a8d60358e74fdf)
+    set(LLVMGenXIntrinsics_GIT_REPO https://github.com/jsji/vc-intrinsics.git)
```
I see the corresponding PR has been merged. Think this is ready for update. Just a reminder. Thanks

jsji replied:
Yes, but I am still waiting for the sync to open source repo. https://github.com/intel/vc-intrinsics

jsji replied:
Finally updated!

asudarsa reviewed (commented):
Test changes in DPC++ Tools area looks good to me.
I do have a request for quick clarification for llvm-spirv changes here. Will it be possible to not make the changes directly here? We should be getting these changes as part of next pulldown from the Khronos SPIRV-LLVM-Translator repo. Adding @MrSidims and @jcranmer-intel for comment here.

Thanks a lot for putting up this major PR.

jsji commented:
> Test changes in DPC++ Tools area looks good to me. I do have a request for quick clarification for llvm-spirv changes here. Will it be possible to not make the changes directly here? We should be getting these changes as part of next pulldown from the Khronos SPIRV-LLVM-Translator repo. Adding @MrSidims and @jcranmer-intel for comment here.
> 
> Thanks a lot for putting up this major PR.

Thanks for the review. The SPIRV changes are here because these are local to `sycl` branch, we have removed them in KhronosGroup/SPIRV-LLVM-Translator repo, but we added these options due to migration, this is to remove them so that we can in sync with KhronosGroup/SPIRV-LLVM-Translator repo again.

jsji commented:
@asudarsa @MrSidims @jcranmer-intel Can you have a look? Hopefully we can get this merged today. :)

asudarsa reviewed (approved):
SPIR-V changes are in sync with open source. So we should be ok.

Thanks

jsji commented:
Thanks @asudarsa ! @intel/llvm-gatekeepers Can we get this merged? Thank you!

jsji commented:
@intel/llvm-gatekeepers Can we get this merged? Thank you!
